src/Advertising.test.js

```diff
@@ -13,7 +18,141 @@ const GPT_SIZE_MAPPING = [
+
+      advertising.setup();
+
+      expect(onErrorSpy).toHaveBeenCalledWith(error);
+    });
+
+    it('should call onError callback when apstag.fetchBids throws an error', async () => {
+      const error = new Error();
+      global.apstag.fetchBids = jest.fn(() => {
+        throw error;
+      });
+
+      await advertising.setup();
+
+      expect(onErrorSpy).toHaveBeenCalledWith(error);
+    });
+
+    it('should call onError callback when apstag.setDisplayBids throws an error', async () => {
+      const error = new Error();
+      global.apstag.fetchBids = jest.fn((_, callback) => {
+        callback();
+      });
+      global.apstag.setDisplayBids = jest.fn(() => {
+        throw error;
+      });
+
+      await advertising.setup();
+
+      expect(onErrorSpy).toHaveBeenCalledWith(error);
+    });
+  });
+
+  describe('call the setup method (without errors)', () => {
+    beforeEach(() => {
+      advertising = new Advertising(config);
+    });
+
+    it('the property `isAPSUsed` should be true', async () => {
+      await advertising.setup();
+
+      expect(advertising.isAPSUsed).toBeTruthy();
+    });
+
+    it('fetchBids and setDisplayBids should not be called if the queue is empty', async () => {
+      await advertising.setup();
+
+      expect(global.apstag.fetchBids).toHaveBeenCalledTimes(0);
+      expect(global.apstag.setDisplayBids).toHaveBeenCalledTimes(0);
+    });
+
+    it('fetchBids and setDisplayBids should be called if there are items in the queue', async () => {
+      advertising.activate(DIV_ID_FOO);
+      await advertising.setup();
+
+      expect(global.apstag.fetchBids).toHaveBeenCalledTimes(1);
+      expect(global.apstag.setDisplayBids).toHaveBeenCalledTimes(1);
+      expect(global.googletag.pubads().refresh).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('call the activate method (with errors)', () => {
+    let onErrorSpy;
+    beforeEach(async () => {
+      onErrorSpy = jest.fn();
+      advertising = new Advertising(config, [], onErrorSpy);
+      await advertising.setup();
+    });
+
+    it('should call onError callback when apstag.fetchBids throws an error', async () => {
+      const error = new Error();
+      global.apstag.fetchBids = jest.fn(() => {
+        throw error;
+      });
+
+      advertising.activate(DIV_ID_FOO);
+
+      expect(onErrorSpy).toHaveBeenCalledWith(error);
+      expect(onErrorSpy).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call onError callback when apstag.setDisplayBids throws an error', async () => {
+      const error = new Error();
+      global.apstag.fetchBids = jest.fn((_, callback) => {
+        callback();
+      });
+      global.apstag.setDisplayBids = jest.fn(() => {
+        throw error;
+      });
+
+      advertising.activate(DIV_ID_FOO);
+
+      expect(onErrorSpy).toHaveBeenCalledWith(error);
+      expect(onErrorSpy).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('call the activate method (without errors)', () => {
```

Comment:
@jmanrumartinez I think we should add a test case where `activate` is called after `setup`, to make sure that `fetchBids` and `setDisplayBids` are called then. I don't believe that's covered by any of the above tests.

Reply:
Done! You're right! I was confused if there's already a test for it, but the above one it's just to test the `.setup` function! 🙌